### PR TITLE
bpo-29694: race condition in pathlib mkdir with flags parents=True

### PR DESCRIPTION
**Pull request by arigo**

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA. This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/test/test_pathlib.py
Comment (serhiy-storchaka) [inline Lib/test/test_pathlib.py]: `assertIn()`/`assertNotIn()`? This will give more information in the case of failure.

```
try:
    ...
except FileExistsError:
    self.assertIn(str(p12), concurrently_created)
else:
    self.assertNotIn(str(p12), concurrently_created)
```

File: Lib/test/test_pathlib.py
Comment (serhiy-storchaka) [inline Lib/test/test_pathlib.py]: Seems this is not used.

File: Misc/NEWS
Comment (serhiy-storchaka) [inline Misc/NEWS]: Add "Fixed" at the start, add a period at the end.

**Comment by serhiy-storchaka** [comment]
In general LGTM, just few minor comments.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

File: Lib/test/test_pathlib.py
Comment (berkerpeksag) [inline Lib/test/test_pathlib.py]: cirCPC looks like a typo here.

File: Misc/NEWS
Comment (berkerpeksag) [inline Misc/NEWS]: You can also add "Patch by Armin Rigo.".

Comment (serhiy-storchaka) [reply Misc/NEWS]: Since Armin is a core developer and will merge the PR himself this is not necessary.

**Comment by arigo** [comment]
I may be a core developer but nowadays I'd call this "historical reasons".  I didn't commit anything for ages.  Please merge this for me.

**Comment by Mariatta** [comment]
I updated Misc/NEWS. I will merge and backport once travis CI is completed.
Thanks :)

**Pull request merged**
